website/docs/cloud-docs/integrations/kubernetes/k8s-ops-v2-migration.mdx

```diff
@@ -0,0 +1,135 @@
+description: >-
+  Upgrade the Terraform Cloud Kubernetes Operator from version 1 to version 2.
+tfc_only: false
+---
+
+# Terraform Cloud Kubernetes Operator v2 Migration Guide
+
+To upgrade the Terraform Cloud Kubernetes Operator from version 1 to version 2, there is a one-time process that you need to complete. This process will upgrade the operator to the newest version and migrate your custom resources. 
+
+## Prerequisites
+
+The migration process requires the following tools to be installed locally:
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- [Helm](https://helm.sh/docs/intro/install/)
+
+## Prepare for the upgrade
+
+Configure an environment variable named `RELEASE_NAMESPACE` with the value of the namespace that the Helm chart is installed in.
+
+```shell-session
+$ export RELEASE_NAMESPACE=<NAMESPACE>
+```
+
+Next, create an environment variable named `RELEASE_NAME` with the value of the name of the installation for your Helm chart.
+
+```shell-session
+$ export RELEASE_NAME=<INSTALLATION_NAME>
+```
+
+Before you migrate to Terraform Cloud Kubernetes Operator v2, you must first update v1 of the operator to the latest version, including the custom resource definitions.
+
+```shell-session
+$ helm upgrade --namespace ${RELEASE_NAMESPACE} ${RELEASE_NAME} hashicorp/terraform
+```
+
+Next, backup the workspace resources.
+
+```shell-session
+$ kubectl get workspace --all-namespaces -o yaml > backup_tfc_operator_v1.yaml
+```
+
+Finally, we suggest that you set the `applyMethod` of each Terraform Cloud workspace to `manual`. In version 2 of the Terraform Cloud Kubernetes Operator this is the default value, however explicitly configuring it with the value provides extra protection from unintended automatic applies during the upgrade process. Run the following command for each Workspace resource.
+
+```shell-session
+kubectl patch workspace <WORKSPACE_NAME> --type=merge --patch '{"spec": {"applyMethod": "manual"}}'
+```
+## Upgrade the operator
+
+Clone the Terraform [Cloud Kubernetes Operator repository](https://github.com/hashicorp/terraform-cloud-operator).
+
+```shell-session
+$ git clone https://github.com/hashicorp/terraform-cloud-operator.git
+```
+
+Change to the migration directory.
+
+```shell-session
+$ cd terraform-cloud-operator/docs/migration/crds
+```
+
+View the changes that **patch A** will apply to the workspace CRD.
+
+```shell-session
+$ kubectl diff -f workspaces_patch_a.yaml
+```
+
+Patch the workspace CRD with **patch A**. This patch adds `app.terraform.io/v1alpha2` support, but excludes `.status.runStatus` because it has a different format in `app.terraform.io/v1alpha1` and causes JSON unmarshalling issues.
+
+```shell-session
+$ kubectl patch crd workspaces.app.terraform.io --patch-file workspaces_patch_a.yaml
+```
+
+Install v2 the Helm chart with the `helm install` command. Be sure to set the `operator.watchedNamespaces` value to the list of namespaces your Workspace resources are deployed to.
```

Comment:
I like that we have mentioned the ability to watch specific namespaces, this is one of the options that practitioners were waiting for. Will it be worth mentioning that without this option the operator will watch all namespaces?

Reply:
Added, thanks!